### PR TITLE
Add hidden options to disable bytes promotion

### PR DESCRIPTION
**Pull request by hauntsaninja**
It might be useful to run mypy_primer without promotions in typeshed. This would give us more confidence in changes stemming from https://github.com/python/typeshed/issues/9001

### CODE REVIEW

File: mypy/main.py [hunk 1]
```diff
@@ -1121,6 +1121,12 @@ def add_invertible_flag(
     parser.add_argument(
         "--enable-incomplete-features", action="store_true", help=argparse.SUPPRESS
     )
+    parser.add_argument(
+        "--_disable-bytearray-promotion", action="store_true", help=argparse.SUPPRESS
```
Comment (JelleZijlstra) [inline mypy/main.py:1125]: I feel like we don't need the underscore if we're already suppressing the flag.

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, great idea!

I think this should be a public option in the future as part of mypy's migration plan to implement PEP 688, if and when it is accepted. But for now, it's useful as an undocumented option for typeshed.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
You need to run Black though.

**Comment by dvarrazzo** [comment]
Looking forward to testing this! Thank you!

**Comment by JelleZijlstra** [comment]
@dvarrazzo fyi, the mypy-primer output in https://github.com/python/mypy/pull/12661#issuecomment-1286429882 should give you an idea of the effect of disabling the promotions on psycopg.

Let me know if you run into any issues where the typeshed stubs don't reflect bytes/bytearray types correctly and I'll help fix them.

**Comment by dvarrazzo** [comment]
@JelleZijlstra I'm afraid I'm not very experienced with running mypy if not from packaged releases.

Which repositories should I check out and install in order to try the feature?

Thank you!

**Comment by JelleZijlstra** [comment]
The option should be in mypy 0.990, which is expected next week.

In the meantime you can simply install from git with `pip install git+https://github.com/python/mypy.git`.

**Comment by hauntsaninja** [comment]
This will be part of 0.990, which should hopefully be out in a week or two: https://github.com/python/mypy/issues/13871

But as of today, `python3 -m pip install -U git+https://github.com/python/mypy.git` is the simplest thing you could do. (There are slightly less simple things you could do if you want the mypyc compiled version of mypy, which is much faster)

**Pull request merged**